pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -593,12 +594,15 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 			}
 			mc.ObjectMeta.UID = uuid.NewUUID()
 			_, ok := cfg.GetAnnotations()[ctrlcommon.MCNameSuffixAnnotationKey]
-			arr := strings.Split(managedKey, "-")
 			// If the MC name suffix annotation does not exist and the managed key value returned has a suffix, then add the MC name
 			// suffix annotation and suffix value to the kubelet config object
-			if len(arr) > 4 && !ok {
```

Comment:
@umohnani8 PTAL. Is the check for len(arr) > 4 necessary? If the custom pool name has multiple `-`, it will also satisfy this case.

Reply:
the check was there because the first MC won't have a number suffix so there would be nothing to add to the annotation. But your patch makes sense, we can get rid of this length check

Reply:
Thanks for the explanation. I keep the check there to protect the `arr[len(arr)-1])` from the panic.